plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -919,8 +919,13 @@ private function build_count_query( $fields, $join, $where, $groupby ) {
 		if ( ! isset( $this->sql ) || '' === $this->sql ) {
 			wc_doing_it_wrong( __FUNCTION__, 'Count query can only be build after main query is built.', '7.3.0' );
 		}
-		$orders_table    = $this->tables['orders'];
-		$this->count_sql = "SELECT COUNT(DISTINCT $fields) FROM  $orders_table $join WHERE $where";
+		$orders_table = $this->tables['orders'];
+		$count_fields = "COUNT(DISTINCT $fields)";
+		if ( "{$orders_table}.id" === $fields && '' === $join ) {
+			// DISTINCT adds performance overhead, exclude the DISTINCT function when confident it is not needed.
+			$count_fields = 'COUNT(*)';
+		}
+		$this->count_sql = "SELECT $count_fields FROM $orders_table $join WHERE $where";
```

Comment:
One minor concern I have with this is that the query can be filtered by `woocommerce_orders_table_query_count_sql` after we remove `DISTINCT` and (in theory) other fields could be added in.

This is probably an edge case, but the safest way might be to remove distinct after that filter if only the `id` field is selected.

Reply:
I thought about that too.  But I only see it being an issue if a filter on `woocommerce_orders_table_query_count_sql` was attempting to `count_sql` string it received rather than regenerate it from scratch, making it even more edge case.  E.g. It would only be an issue if the callback was trying to find/replace within the count_query instead of a having something like:

```
return "SELECT COUNT(DISTINCT $fields) FROM  $orders_table $custom_edited_join WHERE $where";
```

Given that the use of distinct has only been around since #35897, I think the likelihood of a conflict is pretty low.  What do you think?

Reply:
I think this is a small enough edge case and the likelihood of encountering it is pretty slow.

I do think there's a chance of filtering the existing string (as opposed to replacing it), especially when it comes to query modifiers.

On the flip side, anyone replacing the entire count query may end up inadvertently keeping `DISTICINT` in the query and not benefit from this performance optimization.  But in those cases, I think those plugin authors are responsible for optimizing their own queries.